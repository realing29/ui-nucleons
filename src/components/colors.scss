$black: #000;
$white: #fff;

// brand
$brand-dark-blue: #002b41;
$brand-deep-blue: #f5f9fc;
$brand-blue: #1f84db;
$brand-dirty: #80b4d2;
$brand-sky: #e4f1f9;
$brand-gray: #afb8bb;

// gray
$gray87: #212121;
$gray76: #3a3a3b;
$gray66: #545455;
$gray54: #757575;
$gray38: #9e9e9e;
$gray24: #c2c2c2;
$gray12: #e0e0e0;
$gray8: #ebebeb;
$gray4: #f5f5f5;
$gray2: #fafafa;

// service
$service-error: #fb3a2f;
$service-success: #66cb2f;
$service-visited-link: #902bd0;
$service-attention: #e9f1f4;
$service-light-red: #feebea;
$service-light-green: #eff9ea;
$service-sand: #f6f5ed;

// additional
$additional-blue-gray: #607d8b;
$additional-red: #d50000;
$additional-pink: #e82e5c;
$additional-purple: #b52ea8;
$additional-deep-purple: #634bdf;
$additional-blue: #2962ff;
$additional-light-blue: #0091ea;
$additional-cyan: #00b8d4;
$additional-brown: #795548;
$additional-deep-orange: #ff7200;
$additional-amber: #ffab00;
$additional-yellow: #ffd600;
$additional-lime: #aeea00;
$additional-light-green: #64dd17;
$additional-green: #00c853;
$additional-teal: #09ab8b;

$colors: (
  'black': $black,
  'white': $white,
  'brand-dark-blue': $brand-dark-blue,
  'brand-deep-blue': $brand-deep-blue,
  'brand-blue': $brand-blue,
  'brand-dirty': $brand-dirty,
  'brand-sky': $brand-sky,
  'brand-gray': $brand-gray,
  'gray87': $gray87,
  'gray76': $gray76,
  'gray66': $gray66,
  'gray54': $gray54,
  'gray38': $gray38,
  'gray24': $gray24,
  'gray12': $gray12,
  'gray8': $gray8,
  'gray4': $gray4,
  'gray2': $gray2,
  'service-error': $service-error,
  'service-success': $service-success,
  'service-visited-link': $service-visited-link,
  'service-attention': $service-attention,
  'service-light-red': $service-light-red,
  'service-light-green': $service-light-green,
  'service-sand': $service-sand,
  'additional-blue-gray': $additional-blue-gray,
  'additional-red': $additional-red,
  'additional-pink': $additional-pink,
  'additional-purple': $additional-purple,
  'additional-deep-purple': $additional-deep-purple,
  'additional-blue': $additional-blue,
  'additional-light-blue': $additional-light-blue,
  'additional-cyan': $additional-cyan,
  'additional-brown': $additional-brown,
  'additional-deep-orange': $additional-deep-orange,
  'additional-amber': $additional-amber,
  'additional-yellow': $additional-yellow,
  'additional-lime': $additional-lime,
  'additional-light-green': $additional-light-green,
  'additional-green': $additional-green,
  'additional-teal': $additional-teal
);
