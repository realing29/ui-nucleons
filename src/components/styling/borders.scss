@import '../colors';

.inner-border-top {
  box-shadow: inset 0 1px 0 $gray12;
}

.inner-border-bottom {
  box-shadow: inset 0 -1px 0 $gray12;
}

.inner-border-y {
  box-shadow: inset 0 1px 0 $gray12, inset 0 -1px 0 $gray12;
}
