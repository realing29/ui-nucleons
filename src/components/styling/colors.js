import { ClassGetter } from './utils';
import { COLORS } from '../constants';
import classes from './colors.scss';

const isColorKey = COLORS.has.bind(COLORS);

export const color = ClassGetter(classes, isColorKey, 'color__');
export const bgColor = ClassGetter(classes, isColorKey, 'bg-color__');
export const stroke = ClassGetter(classes, isColorKey, 'stroke__');
export const fill = ClassGetter(classes, isColorKey, 'fill__');
