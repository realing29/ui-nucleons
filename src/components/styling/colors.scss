@import '../colors';

@each $key, $color in $colors {
  .color__#{$key} {
    color: $color;
  }
  .background-color__#{$key} {
    background-color: $color;
  }
}
