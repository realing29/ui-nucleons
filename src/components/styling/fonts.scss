$sizes: (
  12,
  14,
  16,
  20,
  24,
  28,
  32,
  48,
  64,
  80,
);

@each $size in $sizes {
  .size-#{$size} {
    font-size: #{$size}px;
  }
  .line-height-#{$size} {
    line-height: #{$size}px;
  }
}

$weights: (
  400,
  500,
  600, // временно нужен для миграции с Open Sans
  700,
);

@each $weight in $weights {
  .weight-#{$weight} {
    font-weight: $weight;
  }
}
