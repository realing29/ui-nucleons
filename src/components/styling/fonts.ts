import { ClassGetter } from './utils';
import classes from './fonts.scss';

export const SIZES = new Set(
  [12, 14, 16, 20, 24, 28, 32, 48, 64, 80]
);

export const WEIGHTS = new Set(
  [400, 500, 600, 700]
);

const isValidSize = SIZES.has.bind(SIZES);

const isValidWeight = WEIGHTS.has.bind(WEIGHTS);

export const size = ClassGetter(classes, isValidSize, 'size-');

export const lineHeight = ClassGetter(classes, isValidSize, 'line-height-');

export const weight = ClassGetter(classes, isValidWeight, 'weight-');
