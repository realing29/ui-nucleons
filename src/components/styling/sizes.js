import allPass from 'lodash/fp/allPass';
import always from 'lodash/fp/always';
import concat from 'lodash/fp/concat';
import __ from 'lodash/fp/__';
import cond from 'lodash/fp/cond';
import gte from 'lodash/fp/gte';
import join from 'lodash/fp/join';
import lte from 'lodash/fp/lte';
import pipe from 'lodash/fp/pipe';
import T from 'lodash/fp/T';
import classnames from 'classnames/bind';

import classes from './sizes.scss';

const isStep = allPass([
  Number.isInteger,
  gte(__, -8),
  lte(__, 32),
]);

const isPositiveStep = allPass([
  Number.isInteger,
  gte(__, 0),
  lte(__, 32),
]);

/**
 * Фабрика геттеров CSS-классов.
 * @param {Function} checker Функция проверки ключа для класса.
 * @param {string} prefix Префикс имени класса.
 * @return {Function} Геттер CSS-класса.
 */
const ClassGetter = (checker, prefix) => cond([
  [checker, pipe([
    concat(prefix),
    join(''),
    classnames.bind(classes),
  ])],
  [T, always(null)],
]);

export const marginTop = ClassGetter(isStep, 'margin-top__');

export const marginRight = ClassGetter(isStep, 'margin-right__');

export const marginBottom = ClassGetter(isStep, 'margin-bottom__');

export const marginLeft = ClassGetter(isStep, 'margin-left__');

export const paddingTop = ClassGetter(isPositiveStep, 'padding-top__');

export const paddingRight = ClassGetter(isPositiveStep, 'padding-right__');

export const paddingBottom = ClassGetter(isPositiveStep, 'padding-bottom__');

export const paddingLeft = ClassGetter(isPositiveStep, 'padding-left__');
