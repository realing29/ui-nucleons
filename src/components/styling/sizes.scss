$step: 4px;

$directions: 'top', 'right', 'bottom', 'left';

// диапазон на основе стилей проектов
@for $index from -8 through 32 {
  @each $direction in $directions {
    .margin-#{$direction}__#{$index} {
      margin-#{$direction}: $step * $index;
    }

    // внутренние отступы могут быть только положительными
    @if $index >= 0 {
      .padding-#{$direction}__#{$index} {
        padding-#{$direction}: $step * $index;
      }
    }
  }
}
