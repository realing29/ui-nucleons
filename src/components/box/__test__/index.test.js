import React from 'react';
import { shallow } from 'enzyme';
import Box from '../index';
import { bgColor } from '../../styling/colors';

describe('<Box />', () => {
  it('should render without props', () => {
    const wrapper = shallow(
      <Box />
    );
    expect(wrapper.find('div')).toHaveLength(1);
  });
  it('should handle "element" prop', () => {
    const wrapper = shallow(
      <Box element='section' />
    );
    expect(wrapper.find('div')).toHaveLength(0);
    expect(wrapper.find('section')).toHaveLength(1);

    wrapper.setProps({ element: undefined });
    expect(wrapper.find('div')).toHaveLength(1);
    expect(wrapper.find('section')).toHaveLength(0);
  });
  it('should handle "children" prop', () => {
    const wrapper = shallow(
      <Box>Test</Box>
    );
    expect(wrapper.find('div').text()).toBe('Test');
  });
  it('should handle "margin*" props', () => {
    const wrapper = shallow(
      <Box
        margin={1}
        marginTop={2}
        marginRight={3}
        marginBottom={4}
        marginLeft={5}
      />
    );
    expect(wrapper.find('div').prop('className')).toBe(
      'margin-top__2 margin-right__3 margin-bottom__4 margin-left__5'
    );

    wrapper.setProps({
      margin: null,
      marginTop: null,
      marginRight: null,
      marginBottom: null,
      marginLeft: null,
      marginX: -6,
      marginY: -7,
    });
    expect(wrapper.find('div').prop('className')).toBe(
      'margin-left__-6 margin-right__-6 margin-top__-7 margin-bottom__-7'
    );

    wrapper.setProps({
      margin: 8,
      marginTop: undefined,
      marginRight: undefined,
      marginBottom: undefined,
      marginLeft: undefined,
      marginX: undefined,
      marginY: undefined,
    });
    expect(wrapper.find('div').prop('className')).toBe(
      'margin-top__8 margin-bottom__8 margin-left__8 margin-right__8'
    );
  });
  it('should handle "padding*" props', () => {
    const wrapper = shallow(
      <Box
        padding={1}
        paddingTop={2}
        paddingRight={3}
        paddingBottom={4}
        paddingLeft={5}
      />
    );
    expect(wrapper.find('div').prop('className')).toBe(
      'padding-top__2 padding-right__3 padding-bottom__4 padding-left__5'
    );

    wrapper.setProps({
      padding: null,
      paddingTop: null,
      paddingRight: null,
      paddingBottom: null,
      paddingLeft: null,
      paddingX: -6,
      paddingY: -7,
    });
    expect(wrapper.find('div').prop('className')).toBe('');

    wrapper.setProps({
      padding: null,
      paddingTop: null,
      paddingRight: null,
      paddingBottom: null,
      paddingLeft: null,
      paddingX: 3,
      paddingY: 2,
    });
    expect(wrapper.find('div').prop('className')).toBe(
      'padding-left__3 padding-right__3 padding-top__2 padding-bottom__2'
    );

    wrapper.setProps({
      padding: 8,
      paddingTop: undefined,
      paddingRight: undefined,
      paddingBottom: undefined,
      paddingLeft: undefined,
      paddingX: undefined,
      paddingY: undefined,
    });
    expect(wrapper.find('div').prop('className')).toBe(
      'padding-top__8 padding-bottom__8 padding-left__8 padding-right__8'
    );
  });
  it('should handle "display" prop', () => {
    const wrapper = shallow(
      <Box />
    );
    const validValues = [
      'block',
      'none',
      'flex',
      'inline',
      'inline-block',
    ];
    validValues.forEach(value => {
      wrapper.setProps({ display: value });
      expect(wrapper.find('div').prop('className')).toContain(`display-${value}`);
    });
    wrapper.setProps({ display: 'invalid' });
    expect(wrapper.find('div').prop('className')).not.toContain('display-invalid');
  });
  it('should handle "flex" prop', () => {
    const wrapper = shallow(
      <Box />
    );
    const validValues = [
      'shrink',
      'grow',
      'none',
    ];
    validValues.forEach(value => {
      wrapper.setProps({ flex: value });
      expect(wrapper.find('div').prop('className')).toContain(`flex-${value}`);
    });
    wrapper.setProps({ flex: 'invalid' });
    expect(wrapper.find('div').prop('className')).not.toContain('flex-invalid');
  });
  it('should handle "wrap" prop', () => {
    const wrapper = shallow(
      <Box />
    );
    expect(wrapper.find('div').prop('className')).not.toContain('flex-wrap');
    wrapper.setProps({ wrap: true });
    expect(wrapper.find('div').prop('className')).toContain('flex-wrap');
  });
  it('should handle "direction" prop', () => {
    const wrapper = shallow(
      <Box />
    );
    const validValues = [
      'row',
      'column',
    ];
    validValues.forEach(value => {
      wrapper.setProps({ direction: value });
      expect(wrapper.find('div').prop('className')).toContain(`direction-${value}`);
    });
    wrapper.setProps({ direction: 'invalid' });
    expect(wrapper.find('div').prop('className')).not.toContain('direction-invalid');
  });
  it('should handle "alignItems" prop', () => {
    const wrapper = shallow(
      <Box />
    );
    const validValues = [
      'start',
      'end',
      'center',
      'baseline',
      'stretch',
    ];
    validValues.forEach(value => {
      wrapper.setProps({ alignItems: value });
      expect(wrapper.find('div').prop('className')).toContain(`align-items-${value}`);
    });
    wrapper.setProps({ alignItems: 'invalid' });
    expect(wrapper.find('div').prop('className')).not.toContain('align-items-invalid');
  });
  it('should handle "justifyContent" prop', () => {
    const wrapper = shallow(
      <Box />
    );
    const validValues = [
      'start',
      'end',
      'center',
      'between',
      'around',
    ];
    validValues.forEach(value => {
      wrapper.setProps({ justifyContent: value });
      expect(wrapper.find('div').prop('className')).toContain(`justify-content-${value}`);
    });
    wrapper.setProps({ justifyContent: 'invalid' });
    expect(wrapper.find('div').prop('className')).not.toContain('justify-content-invalid');
  });
  it('should handle "color" prop', () => {
    const wrapper = shallow(
      <Box color='service-error' />
    );

    expect(wrapper.find('div').prop('className')).toContain(bgColor('service-error'));
  });
});
