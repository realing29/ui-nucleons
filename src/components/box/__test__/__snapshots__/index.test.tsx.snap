// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> should handle "dangerouslySetInlineStyle" prop 1`] = `
<Box
  dangerouslySetInlineStyle={
    Object {
      "__style": Object {
        "textAlign": "left",
      },
    }
  }
>
  <div
    className=""
    style={
      Object {
        "textAlign": "left",
      }
    }
  />
</Box>
`;
